AdyenCardScanner/Sources/CardScannerViewModel.swift

```diff
@@ -0,0 +1,259 @@
+        self.previewFrame = previewLayerFrame
+        self.roiInPreviewFrame = roiInPreviewLayer
+    }
+
+    func updateVideoOrientation() {
+        guard
+            let connection = videoPreviewLayer.connection,
+            connection.isVideoOrientationSupported else {
+            return
+        }
+
+        connection.videoOrientation = .currentVideoOrientation
+        videoPreviewLayer.removeAllAnimations()
+    }
+
+    // MARK: - Private
+
+    private func configureCaptureSession() {
+        captureSession.beginConfiguration()
+
+        // Set up input
+        guard
+            let videoInput = try? AVCaptureDeviceInput(device: captureDevice),
+            captureSession.canAddInput(videoInput)
+        else {
+            let cardScannerError = CardScannerError(kind: .capture)
+            completion(.failure(cardScannerError))
+            return
+        }
+
+        // Configure capture device settings
+        configureCaptureDevice(captureDevice)
+
+        captureSession.addInput(videoInput)
+
+        // Set up output
+        let videoOutput = AVCaptureVideoDataOutput()
+        let videoSettings = [kCVPixelBufferPixelFormatTypeKey as NSString: NSNumber(value: kCVPixelFormatType_32BGRA)] as [String: Any]
+        videoOutput.videoSettings = videoSettings
+        videoOutput.setSampleBufferDelegate(self, queue: videoOutputQueue)
+
+        guard captureSession.canAddOutput(videoOutput) else {
+            let cardScannerError = CardScannerError(kind: .capture)
+            completion(.failure(cardScannerError))
+            return
+        }
+
+        captureSession.addOutput(videoOutput)
+
+        guard
+            let connection = videoOutput.connection(with: .video),
+            connection.isVideoStabilizationSupported else {
+            let cardScannerError = CardScannerError(kind: .capture)
+            completion(.failure(cardScannerError))
+            return
+        }
+        connection.preferredVideoStabilizationMode = .auto
+
+        captureSession.commitConfiguration()
+    }
+
+    private func configureCaptureDevice(_ device: AVCaptureDevice) {
+        try? device.lockForConfiguration()
+
+        // Adjust Frame Rate
+        device.activeVideoMinFrameDuration = CMTime(value: 1, timescale: 15) // 15 fps
+        device.activeVideoMaxFrameDuration = CMTime(value: 1, timescale: 30) // 30 fps
+
+        // Focus and Exposure Settings
+        if device.isFocusModeSupported(.continuousAutoFocus) {
+            device.focusMode = .continuousAutoFocus
+        }
+
+        // Auto Exposure
+        if device.isExposureModeSupported(.continuousAutoExposure) {
+            device.exposureMode = .continuousAutoExposure
+        }
+
+        device.unlockForConfiguration()
+    }
+
+    private func getCardData(
+        from image: CIImage,
+        roiInPreviewFrame: CGRect,
+        previewFrame: CGRect
+    ) {
+        guard let croppedImage = cropRegionOfInterest(
```

Comment:
Could be nice to move this into CIImage extension. Also can simplify `previewFrame` to just `CGSize`

```suggestion
        guard let croppedImage = image. crop(roiFrame: CGRect, previewSize: CGSize)
```

Reply:
For now I want to keep this method here as its logic its very specific to the view model